### ISSUE TITLE
Rerun results are not being updated in existing test instead creating new test in same launch[v5]

### ISSUE THREAD

**Issue by VenkatVanacharlaTR** [reportportal/reportportal#1362]
Describe the bug.
Recently we upgraded Report Portal UI from 5.3.3 to 5.3.5.
After upgrading rerun results are not getting updated for launches created using Java TestNG Framework but added as a new tests in the same launch which indirectly affecting our total test count and pass percentage.

To Reproduce
1. Take TestNG xml and run with rp.rerun = false.
2. then remove few tests/reorder the tests and run with rp.rerun = true
Facing this issue consistently

Expected behavior
Rerun results need to updated for all rerun tests.

Actual:
Rerun tests are added as new tests in another test path.

Ivy dependencies:
agent-java-testng 5.0.9-TESTNG6
logger-java-log4j 5.0.3

![RP Rerun issue](https://user-images.githubusercontent.com/76486046/114344452-7b7dee80-9b7d-11eb-9a3d-90ada82be68e.PNG)

**Comment by VenkatVanacharlaTR** [comment]
@evjlobanova, Could you please let me know when it will going to be fixed?

**Comment by evjlobanova** [comment]
Bug with rerun would be fixed in version 5.4.1.
±ETA end of the June

**Comment by TAVkiev** [comment]
@evjlobanova hi, new version released - 5.5. Still rerun function creates duplicates of tests

**Comment by creatorOfautomation** [comment]
@evjlobanova Same for me. Mostly it works but it doesn't work for the following:

1. For the failed @Before and @After classes/method - results are not being updated in existing item instead creating new item in a new Suite in a Launch
2. For failed tests with parameters obtained from @DataProvider - results are not being updated in existing test instead creating new test. It's happening because @DataProvider contains random data. As far as I understood when reruning such tests those TC Id differ from the TC Id of the main lauch 

RP - 5.5, agent-java-testng - 5.0.11

**Comment by TAVkiev** [comment]
@evjlobanova I reproduced it with js agent. Issue also created, all data that I used from launch to launch - static, I set it only directly only once
https://github.com/reportportal/agent-js-mocha/issues/67
